Feature/shutter control

coeing opened the pull request:
TODO:
- [x] Stopping not working as implemented, check if possible or remove
- [x] Add PercentType command to set precise percentage of shutter level
- [x] Move service name "ShutterControl" into constant variable
- [x] Remove Gateway-ID from header of putState request

<!--
Thanks for contributing to the openHAB project!
Please describe the goal and effect of your PR here.
Pay attention to the below notes and to *the guidelines* for this repository.
Feel free to delete any comment sections in the template (starting with "<!--").
-->

<!-- TITLE -->

<!--
Please provide a PR summary in the *Title* above, according to the following schema:
- If related to one specific add-on: Mention the add-on shortname in square brackets
  e.g. "[exec]", "[netatmo]" or "[tesla]"
- If the PR is work in progress: Add "[WIP]"
- Give a short meaningful description in imperative mood
  e.g. "Add support for device XYZ" or "Fix wrongly handled exception"
  for a new add-on/binding: "Initial contribution"
Examples:
- "[homematic] Improve communication with weak signal devices"
- "[timemachine][WIP] Initial contribution"
- "Update contribution guidelines on new signing rules"
-->

<!-- DESCRIPTION -->

<!--
Please give a few sentences describing the overall goals of the pull request.
Give enough details to make the improvement and changes of the PR understandable
to both developers and tech-savy users.

Please keep the following in mind:
- What is the classification of the PR, e.g. Bugfix, Improvement, Novel Addition, ... ?
- Did you describe the PRs motivation and goal?
- Did you provide a link to any prior discussion, e.g. an issue or community forum thread?
- Did you describe new features for the end user?
- Did you describe any noteworthy changes in usage for the end user?
- Was the documentation updated accordingly, e.g. the add-on README?
- Does your contribution follow the coding guidelines:
  https://www.openhab.org/docs/developer/development/guidelines.html
- Did you check for any (relevant) issues from the static code analysis:
  https://www.openhab.org/docs/developer/development/bindings.html#static-code-analysis
- Did you sign-off your work:
  https://www.openhab.org/docs/developer/contributing/contributing.html#sign-your-work
-->

<!-- TESTING -->

<!--
Your Pull Request will automatically be built and available under the following folder:
https://openhab.jfrog.io/openhab/libs-pullrequest-local/org/openhab/

It is a good practice to add a URL to your built JAR in this Pull Request description,
so it is easier for the community to test your Add-on.
If your Pull Request contains a new binding, it will likely take some time
before it is reviewed and processed by maintainers.
That said, consider submitting your Add-on in the Eclipse IoT Marketplace
See this thread for more info:
https://community.openhab.org/t/24491

Don't forget to submit a thread about your Add-on in the openHAB community:
https://community.openhab.org/c/add-ons 
-->

stefan-kaestle commented on bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/BoschSHCBridgeHandler.java:
```diff
@@ -537,9 +540,39 @@ private Boolean getRooms() {
+        Gson gson = new Gson();
+        String body = gson.toJson(state);
+        Request request = this.httpClient.newRequest(url).header("Content-Type", "application/json")
+                .header("Accept", "application/json").header("Gateway-ID", "64-DA-A0-02-14-9B").method(PUT)
```
There is a hard-coded MAC address in here :-)
Not sure if that's a problem?

stefan-kaestle replied:
I think I have them in various places too :-P

stefan-kaestle commented on bundles/org.openhab.binding.boschshc/.classpath:
```diff
@@ -28,5 +28,19 @@
 			<attribute name="maven.pomderived" value="true"/>
 		</attributes>
 	</classpathentry>
+	<classpathentry kind="src" output="target/test-classes" path="target/generated-test-sources/test-annotations">
```
Not sure what this thing does? :-)

coeing commented:
Stopping the shutters from moving doesn't seem possible with the current API, so I created an issue for it: https://github.com/BoschSmartHome/bosch-shc-api-docs/issues/34 

Maybe I can find a workaround in the meantime.

coeing commented:
Alright, this was a short but very successful coding session :) I handled all the remaining TODOs, so from my side the pull request is ready to merge. But please check if it is okay from your side as well!

I am using some general functions that can also be used in other places, but I would like to do the refactoring in a separate branch.

Next time I have time I will add some issues for the parts I'd like to change, so we can discuss them if necessary.

coeing commented:
@stefan-kaestle Thanks for the review :) 

About your points:
- I can add some information about the new supported thing to the Readme.md, that's a good point!
- I would suggest to have a list of contributors in the Readme.md and not in every single code file. If anybody is interested who created which code fragment, he/she could easily use Git Blame for it :) 

If you give me write access to the repository, I'll merge my branch after the addition of the shutter control info to the Readme.md. Otherwise I'll let you know when I did the addition :)

coeing commented:
Alright, I added the Shutter Control device to the supported device section of the Readme.md and added a null reference check because on initialization the device state doesn't seems to be fetched correctly.

Will merge the pull request now! :scream:

stefan-kaestle commented:
Great work! Thanks Christian! :tada::tada::tada::tada:
